Sources/IdentityState.swift

```diff
@@ -31,7 +31,7 @@ class IdentityState {
 
     /// Completes init for this Identity extension.
     /// - Returns: True if we should share state after bootup, false otherwise
```

Comment:
It is a bit confusing on when bootupIfReady returns true/false, it looks like we are using it to determine if it booted and also if shared state needs to be updated or not. 
We can consolidate that by doing the following changes:
1. `if hasBooted { return true }`
2. create shared state inside IdentityState when bootup is completed
3. simplify readyForEvent

Reply:
I've added `createXDMSharedState` as a parameter to `bootupIfReady` and changed the return value to be true/false if the bootup completed successfully. I also added more comments to document the boot up process.